install-npm.js

```diff
@@ -4,8 +4,14 @@ var npmModule
   , path = require('path')
   ;
 
-if (process.env['npm_execpath'] && process.env['npm_execpath'].match(/\/node_modules\/npm\/bin\/npm-cli\.js$/)) {
-  npmModule = require(path.resolve(process.env['npm_execpath'], '..', '..'));
+if (process.env['npm_execpath']) {
+  if (process.platform === 'win32') {
+    process.env['npm_execpath'] = process.env['npm_execpath'].replace(/\\/g, '/');
+  }
+
+  if (process.env['npm_execpath'].match(/\/node_modules\/npm\/bin\/npm-cli\.js$/)) {
```

Comment:
I think we can utilize built-in node functionality in handling platform FS differences
```suggestion
  var expectedNpmPath = path.join('bin', 'npm-cli.js')
  if (process.env['npm_execpath'].substr(-1 * expectedNpmPath.length) === expectedNpmPath) {
```

Reply:
Latest approach uses path.join to build a path to both NPM and Yarn binaries, then compares to execpath.